File: 55-jump-game/55-jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) 
    {
        if (nums.size() == 1) return true;
        int end = nums[0];
        
        for (int i = 0; i < nums.size() && i <= end; i++)
        {
            if (nums[i] + i >= nums.size() - 1) return true;
                
            end = max(i + nums[i], end);
        }
        
        return false;
    }
};

File: 55-jump-game/55-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:        
        count = 0
        if nums[-1] == 0:
            count = -1
        for i in range(-1,-(len(nums)+1),-1):
            if nums[i] <= count:
                count+=1
            else:
                count = 0
        return count == 0 
        